src/ol/control/fullscreencontrol.js

```diff
@@ -111,7 +121,8 @@ ol.control.FullScreen.prototype.handleFullScreen_ = function() {
   if (goog.dom.fullscreen.isFullScreen()) {
     goog.dom.fullscreen.exitFullScreen();
   } else {
-    var element = map.getTargetElement();
+    var element = this.fullScreenSource_ !== undefined ?
+      goog.dom.getElement(this.fullScreenSource_) : map.getTargetElement();
```

Comment:
A truthy check should suffice here.  And we typically do two indents for multiline ternary expressions (I'll see if I can configure the linter to enforce this).

So, this could look like:
```
var element = this.fullScreenSource_ ?
    goog.dom.getElement(this.fullScreenSource_) : map.getTargetElement();
```